### PR TITLE
FIX Fix feature detection when `obj.constructor` is undefined

### PR DESCRIPTION
**Pull request by hoodmane**
Currently the following code fails:
```py
from js import eval
eval("Object.create(null)")
```
with:
```py
Traceback (most recent call last):
  File "<console>", line 1, in <module>
JsException: TypeError: Cannot read properties of undefined (reading 'name')
```
This fixes it.

- [x] Update tests
- [x] Update changelog

### CODE REVIEW

**Review by ryanking13 — COMMENTED** [review]
Wow thanks, @hoodmane. Nice catch.

```python
>>> from js import eval
>>> nullobj = eval("Object.create(null)")
>>> nullobj
Traceback (most recent call last):
  File "/lib/python3.10/site-packages/pyodide/console.py", line 489, in repr_shorten
    text = repr(value)
  File "<omitted>", line 102, in S.fn.init.interpreter
JsException: TypeError: Hiwire.get_value(...).toString is not a function
>>> repr(nullobj)
Pyodide has suffered a fatal error. Please report this to the Pyodide maintainers.
The cause of the fatal error was:
TypeError: Cannot read properties of undefined (reading 'fileName')
Look in the browser console for more details.
```

I'm getting this error when I try to print null object.

**Comment by hoodmane** [comment]
null objects are problematic in a lot of ways... I ran into it in https://github.com/pyodide/pyodide/issues/2519#issuecomment-1120285958

File: src/core/jsproxy.c [hunk 1]
```diff
@@ -119,6 +119,12 @@ static PyObject*
 JsProxy_Repr(PyObject* self)
 {
   JsRef idrepr = hiwire_to_string(JsProxy_REF(self));
+  if (idrepr == NULL) {
+    PyErr_Format(PyExc_TypeError,
+                 "Pyodide cannot generate a repr for this Javascript object "
+                 "because it has no 'toString' method");
+    return NULL;
```
Comment (ryanking13) [inline src/core/jsproxy.c:126]: Yes, I think this is reasonable as `alert(Object.create(null))` also raises TypeError in the browser.

**Review by ryanking13 — APPROVED** [review]
Thanks for work @hoodmane!

**Comment by hoodmane** [comment]
Thanks for the review @ryanking13!

**Pull request merged**